### DISCUSSION TITLE
Customizable Log Output

### DISCUSSION THREAD

**Discussion by tiredofit** [stalwartlabs/smtp-server#37] [Q&A]
I am trying to match or exceed functionality from a long running Postfix MTA. One of the changes I have made is to track the subject within the log file. Are there plans in the future to add custom logging functionality?

**Comment by mdecimus** [comment]
Currently Stalwart supports different logging mechanisms, but what do you mean exactly by custom logging functionality?

**Reply by tiredofit** [reply]
I realize my above use case wasn't entirely clear. I would like to track the subject in the log file, or another use case (which may already exist, I just haven't been able to get it to happen) would be to output the spamfilter score within the log as well.

**Reply by mdecimus** [reply]
This is quite specific to be added as a feature but this could be implemented by adding logging capabilities within Sieve. This way you could log any information you like at the `DATA` stage of the SMTP transaction.
